core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -109,20 +109,17 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
 
       new Thread(Utils.getFormattedClassName(this) + "-mesos-driver") {
         setDaemon(true)
+        setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler {
```

Comment:
This pattern is used in many other places in Spark though but if it helps I can change it back to try-catch.